avl-tree/src/lib.rs

```diff
@@ -0,0 +1,589 @@
+use bytemuck::{Pod, Zeroable};
+use node_allocator::{NodeAllocator, ZeroCopy, SENTINEL};
+use std::{
+    cmp::max,
+    ops::{Index, IndexMut},
+};
+
+// The number of registers.
+const REGISTERS: usize = 3;
```

Comment:
I think there probably needs to be 4 registers in order for bytemuck to properly align the bytes on 64-bit systems

Reply:
Right, I will change that. I will refactor the code to use this "extra" register to store the parent reference and simplify the logic to traverse the tree.

Reply:
I increased the number of registers and considered storing the parent reference in it. Found that there is a trade-off between adding the logic to store/update the parent reference and the current logic to traverse the tree without it. In the end, I am not sure whether there is a significant difference/gain or not, so decided not to change it.

Reply:
Storing the parent would be a nice touch, but I don't think it's strictly needed for this implementation. Feel free to leave it as a TODO. Awesome work!

Reply:
Thanks! I added a TODO note about the possibility of using the extra register to store the parent reference.